### PR TITLE
FIX: Composite types missing in context menu when "Any" ControlType selected (ISXB-769)

### PR DESCRIPTION
**Pull request by timkeo**
### Description

Simple fix to the filtering logic of CompostieTypes added to the context menu.

### Changes made

Changed the condition for "Any" to check for a null string.

From my investigation, this functionality was refactored and improved [in this PR](https://github.com/Unity-Technologies/InputSystem/pull/1757), but it exposed [another bug]([ISX-1650](https://jira.unity3d.com/browse/ISX-1650)), which was fixed by [this other PR](https://github.com/Unity-Technologies/InputSystem/pull/1788).

As I understand it, the 2nd PR changed the handling of the "Any" ControlType, so it's always index 0 and doesn't have a set value, i.e. expected to be null. However, the condition for the context menu was missed with this change.

### Notes

I'm not familiar with the expected CompositeTypes for "Any". I tried to compare with IMGUI implementation in 1.7 version, both the ControlTypes seemed to have changed, and there's now a "Custom" CompositeType in current version. So it's not a 1-1 comparison, and hopefully this is the correct functionality.

### Checklist

Before review:

- [x] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - FogBugz ticket attached, example `([case %number%](https://issuetracker.unity3d.com/issues/...))`.
    - FogBugz is marked as "Resolved" with *next* release version correctly set.
- [ ] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [ ] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

During merge:

- [x] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

### CODE REVIEW

**Review by ekcoh — APPROVED** [review]
Fix LGTM, not sure I understand what is asked for in the PR description but Any basically removes any constraint to type which means action may consume object type and basically consumer of action needs to type check and cast at use.

File: Packages/com.unity.inputsystem/CHANGELOG.md [hunk 1]
```diff
@@ -65,6 +65,8 @@ however, it has to be formatted properly to pass verification tests.
 - Fixed an issue where selection state of UI Toolkit editor state would not be preserved when associated with a new serialized copy of the asset.
 - Fixed an issue where any exceptions throw from within UI Toolkit event queue would only log the error message and not the full exception stack trace, making debugging more difficult.
 - Fixed an issue where UI Toolkit Input Action Editor wouldn't provide a correct modification state when coming back from domain reload.
+- Fixed duplication of project wide input actions when loading/unloading scenes.
```
Comment (lyndon-unity) [inline Packages/com.unity.inputsystem/CHANGELOG.md]: Is this line about duplication a merge error. Looks like its unrelated to the fix.

**Review by lyndon-unity — APPROVED** [review]
Change looks reasonable so approving but the change log may have a unnecessary entry.

File: Packages/com.unity.inputsystem/CHANGELOG.md (hunk 1 above)
Comment (ekcoh) [reply Packages/com.unity.inputsystem/CHANGELOG.md]: I have cleaned this up when merging latest develop into this branch, there was a bad merge conflict resolution introducing a duplicate.

**Comment by Pauliusd01** [comment]
Hey Tim, I think we were missing these additional composite types from the video and your PR correctly adds them back. But the issue is that they are not immediately available in the Any Control Type so a switch to a different one and back is needed for them to appear. Could you make it so they're properly available right away?

https://github.com/Unity-Technologies/InputSystem/assets/54306142/716c0be6-6a98-4017-9749-b415f573d744

**Comment by timkeo** [comment]
> Hey Tim, I think we were missing these additional composite types from the video and your PR correctly adds them back. But the issue is that they are not immediately available in the Any Control Type so a switch to a different one and back is needed for them to appear. Could you make it so they're properly available right away?
> Unity_2024-03-04_11-02-39.mp4

I quickly investigated this issue and discovered it's another, separate bug and opened [ISX-1916](https://jira.unity3d.com/browse/ISX-1916) for it.

**Pull request merged**